lib/libvfio-user.c

```diff
@@ -1266,6 +1303,12 @@ vfu_setup_region(vfu_ctx_t *vfu_ctx, int region_idx, size_t size,
         return ERROR(EINVAL);
     }
 
+    if (region_idx == VFU_PCI_DEV_MIGR_REGION_IDX &&
+        size < sizeof(struct vfio_device_migration_info)) {
```

Comment:
This should be requiring the page aligned size (i.e. what your utility function returns), not the struct size. Otherwise it's just more confusing I think.